clientgui/AsyncRPC.cpp

```diff
@@ -159,7 +161,10 @@ void *RPCThread::Entry() {
         // This does the following:
         // (1) Unlocks the Mutex and puts the RPC thread to sleep as an atomic operation.
         // (2) On Signal from main thread: locks Mutex again and wakes the RPC thread.
-        condErr = m_pRPC_Thread_Condition->Wait();
+#if wxDEBUG_LEVEL
+        condErr =
```

Comment:
Maybe it would be better to check these function calls results in both Debug and Release? Just ignoring return value of async functions might lead to application crash or deadlock that is not nice either. And having `wxASSERT` here is not a solution either because such an errors might happen on Release only code when it's optimized (assuming that we're using at least 3 compilers (Microsoft cl for Windows, gcc for linux and clang for OSX)) that in result might produce the code that have different behavior on every platform.
So here I'd personally suggest to rewrite this function and make proper error handling instead of just ignoring function calls results in Release mode.

Reply:
The problem with your suggestion is that I have no idea what we could do in production if the call results in an error. So the asserts are there for debugging purposes. You definitely don't want a production app to crash unnecessarily, which is what an assert appears to do when not run under a debugger.

This code was very difficult to get working originally, and I have no interest in doing major work on it. In any case, what you are suggesting is well beyond the scope of this PR. If you want, please create a separate issue for it.

Reply:
Ok, I agree with you. Let's keep it then 'as-is' for now. I'll put this in my 'nice to do' list and probably handle this properly one day.